src/main/java/org/folio/event/handler/HoldingCreateAsyncRecordHandler.java

```diff
@@ -0,0 +1,120 @@
+          processPiecesUpdate(holdingId, tenantId, headers, conn)
+        );
+        return GenericCompositeFuture.all(tenantIdUpdates).mapEmpty();
+      });
+  }
+
+  private Future<Void> processPoLinesUpdate(String holdingId, String tenantId, Map<String, String> headers, Conn conn) {
+    return poLinesService.getPoLinesByHoldingId(holdingId, conn)
+      .compose(poLines -> updatePoLines(poLines, holdingId, tenantId, conn))
+      .compose(poLines -> auditOutboxService.saveOrderLinesOutboxLogs(conn, poLines, OrderLineAuditEvent.Action.EDIT, headers))
+      .onSuccess(ar -> auditOutboxService.processOutboxEventLogs(headers))
+      .mapEmpty();
+  }
+
+  private Future<Void> processPiecesUpdate(String holdingId, String tenantId, Map<String, String> headers, Conn conn) {
+    return pieceService.getPiecesByHoldingId(holdingId, conn)
+      .compose(pieces -> updatePieces(pieces, holdingId, tenantId, conn))
+      .compose(pieces -> auditOutboxService.savePiecesOutboxLog(conn, pieces, PieceAuditEvent.Action.CREATE, headers))
+      .onSuccess(ar -> auditOutboxService.processOutboxEventLogs(headers))
+      .mapEmpty();
+  }
+
+  private Future<List<PoLine>> updatePoLines(List<PoLine> poLines, String holdingId, String tenantId, Conn conn) {
+    if (CollectionUtils.isEmpty(poLines)) {
+      log.info("updatePoLines:: No poLines to update for holding: '{}' and tenant: '{}'", holdingId, tenantId);
+      return Future.succeededFuture();
+    }
+    log.info("updatePoLines:: Updating {} poLine(s) with holdingId '{}', setting receivingTenantId to '{}'", poLines.size(), holdingId, tenantId);
+    poLines.forEach(poLine -> updateLocationTenantIdIfNeeded(poLine.getLocations(), holdingId, tenantId));
+    return poLinesService.updatePoLines(poLines, conn, tenantId)
+      .map(v -> poLines);
+  }
+
+  private Future<List<Piece>> updatePieces(List<Piece> pieces, String holdingId, String tenantId, Conn conn) {
+    var piecesToUpdate = pieces.stream()
+      .filter(piece -> !Objects.equals(piece.getReceivingTenantId(), tenantId))
+      .map(piece -> piece.withReceivingTenantId(tenantId))
+      .toList();
+    if (CollectionUtils.isEmpty(piecesToUpdate)) {
+      log.info("updatePieces:: No pieces to update for holding: '{}' and tenant: '{}'", holdingId, tenantId);
+      return Future.succeededFuture();
+    }
+    log.info("updatePieces:: Updating {} piece(s) with holdingId '{}', setting receivingTenantId to '{}'", pieces.size(), holdingId, tenantId);
+    return pieceService.updatePieces(piecesToUpdate, conn, tenantId);
+  }
+
+  private void updateLocationTenantIdIfNeeded(List<Location> locations, String holdingId, String tenantId) {
+    locations.stream()
+      .filter(location -> Objects.equals(location.getHoldingId(), holdingId))
+      .forEach(location -> location.setTenantId(tenantId));
+  }
+
+  @Override
+  protected Future<Optional<ConsortiumConfiguration>> getConsortiumConfiguration(Map<String, String> headers) {
```

Comment:
this method can be just in base class, because it has the same implementation in subclasses